gui/quantum.lua

```diff
@@ -11,262 +11,218 @@ local quickfort = reqscript('quickfort')
 local quickfort_command = reqscript('internal/quickfort/command')
 local quickfort_orders = reqscript('internal/quickfort/orders')
 
-QuantumUI = defclass(QuantumUI, guidm.MenuOverlay)
+
+-- UI Layout
+QuantumUI = defclass(QuantumUI, gui.ZScreen)
 QuantumUI.ATTRS {
-    frame_inset=1,
     focus_path='quantum',
-    sidebar_mode=df.ui_sidebar_mode.LookAround,
+    pass_movement_keys=true,
+    pass_mouse_clicks=false
 }
 
-function QuantumUI:init()
-    local cart_count = #assign_minecarts.get_free_vehicles()
+QuantumWin = defclass(QuantumWin, widgets.Window)
+QuantumWin.ATTRS {
+    frame_title = 'Quantum Stockpile',
+    frame = { w=60, h=26 },
+    resizable = true,
+    autoarrange_subviews=true
+}
 
-    local main_panel = widgets.Panel{autoarrange_subviews=true,
-                                     autoarrange_gap=1}
-    main_panel:addviews{
-        widgets.Label{text='Quantum'},
+function QuantumWin:init()
+    cart_count = #assign_minecarts.get_free_vehicles()
+
+    self:addviews{
+        widgets.Label{
+            text='Quantum'
+        },
         widgets.WrappedLabel{
             text_to_wrap=self:callback('get_help_text'),
-            text_pen=COLOR_GREY},
-        widgets.ResizingPanel{autoarrange_subviews=true, subviews={
-            widgets.EditField{
-                view_id='name',
-                key='CUSTOM_N',
-                on_char=self:callback('on_name_char'),
-                text=''},
-            widgets.TooltipLabel{
-                text_to_wrap='Give the quantum stockpile a custom name.',
-                show_tooltip=true}}},
-        widgets.ResizingPanel{autoarrange_subviews=true, subviews={
-            widgets.CycleHotkeyLabel{
-                view_id='dir',
-                key='CUSTOM_D',
-                options={{label='North', value={y=-1}},
-                         {label='South', value={y=1}},
-                         {label='East', value={x=1}},
-                         {label='West', value={x=-1}}}},
-            widgets.TooltipLabel{
-                text_to_wrap='Set the dump direction of the quantum stop.',
-                show_tooltip=true}}},
-        widgets.ResizingPanel{autoarrange_subviews=true, subviews={
-            widgets.ToggleHotkeyLabel{
-                view_id='refuse',
-                key='CUSTOM_R',
-                label='Allow refuse/corpses',
-                initial_option=false},
-            widgets.TooltipLabel{
-                text_to_wrap='Note that enabling refuse will cause clothes' ..
+            text_pen=COLOR_GREY
+        },
+        widgets.HotkeyLabel{
+            key='CUSTOM_Q',
+            label='Place QSP under cursor',
+            on_activate=self:callback('commit')
+        },
```

Comment:
I'd say don't bother with the keyboard cursor for now. Few people know how to use it and it complicates the UI. Go with mouse clicks, like `gui/quantum` and `gui/blueprint` and other map tools do

Reply:
I had left the keyboard option in cause there's a part o the community that gets cranky when there's not keyboard shortcuts for stuff. The Q key places it where the mouse cursor is, so I can see why it would be redundant, and I understand it not being worth it to cater to that small subsection of the community.